src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpSystemProxy.cs

```diff
@@ -131,8 +156,59 @@ public Uri GetProxy(Uri uri)
         {
             if (_proxyHelper.ManualSettingsOnly)
             {
+                if ( _bypassLocal)
+                {
+                    IPAddress address = null;
+
+                    if (uri.IsLoopback)
+                    {
+                        // This is optimization for loopback addresses.
+                        // Unfortunately this does not work for all local addresses.
+                        return null;
+                    }
+                    if (uri.Host[0] == '[' || Char.IsNumber(uri.Host[0]))
```

Comment:
Nit: please add a blank line before each of these if blocks

Reply:
Char.IsNumber allows anything in the unicode number category... is that what's desired here?

Reply:
Is it possible Host.Length could be 0?

Reply:
And is Host correct to be using here?  Or IdnHost? (I don't know the answer.)

Reply:
I check and we will get punicoded string from system. However Idn does not have [] for IPv6 as well as wildcards are more tricky. 
I did testing with WinHttp and the bypass seems broken even for exact match.